### PR TITLE
Database resource to framework + Grafana!

### PR DESCRIPTION
**Pull request by kobajagi**
This PR moves migrates database resource and database URI datasource to terraform-plugin-framework. Also adds Grafana database support.

```
$ TF_ACC=1 go test ./... -v -timeout=60m -run=^TestDatabase 
?       github.com/exoscale/terraform-provider-exoscale [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/config      [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/general     [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/list        [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/provider    [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/provider/config     [no test files]
testing: warning: no tests to run
PASS
ok      github.com/exoscale/terraform-provider-exoscale/exoscale        0.008s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/exoscale/terraform-provider-exoscale/pkg/filter      (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/exoscale/terraform-provider-exoscale/pkg/resources/anti_affinity_group       0.008s [no tests to run]
?       github.com/exoscale/terraform-provider-exoscale/pkg/resources/zones     [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/testutils   [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/utils       [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/validators  [no test files]
?       github.com/exoscale/terraform-provider-exoscale/pkg/version     [no test files]
?       github.com/exoscale/terraform-provider-exoscale/version [no test files]
=== RUN   TestDatabase
=== RUN   TestDatabase/ResourcePg
=== RUN   TestDatabase/ResourceMysql
=== RUN   TestDatabase/ResourceRedis
=== RUN   TestDatabase/ResourceKafka
=== RUN   TestDatabase/ResourceOpensearch
=== RUN   TestDatabase/ResourceGrafana
=== RUN   TestDatabase/DataSourceURI
--- PASS: TestDatabase (140.45s)
    --- PASS: TestDatabase/ResourcePg (12.25s)
    --- PASS: TestDatabase/ResourceMysql (15.26s)
    --- PASS: TestDatabase/ResourceRedis (11.57s)
    --- PASS: TestDatabase/ResourceKafka (17.87s)
    --- PASS: TestDatabase/ResourceOpensearch (18.07s)
    --- PASS: TestDatabase/ResourceGrafana (17.20s)
    --- PASS: TestDatabase/DataSourceURI (48.23s)
PASS
ok      github.com/exoscale/terraform-provider-exoscale/pkg/resources/database  140.463s
testing: warning: no tests to run
PASS
ok      github.com/exoscale/terraform-provider-exoscale/pkg/resources/instance  0.006s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/exoscale/terraform-provider-exoscale/pkg/resources/instance_pool     0.006s [no tests to run]
```

### CODE REVIEW

File: pkg/resources/database/resource_opensearch.go [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+	Enabled         types.Bool  `tfsdk:"enabled"`
+	MaxOldSpaceSize types.Int64 `tfsdk:"max_old_space_size"`
+	RequestTimeout  types.Int64 `tfsdk:"request_timeout"`
+}
+
+var ResourceOpensearchSchema = schema.SingleNestedBlock{
+	MarkdownDescription: "*opensearch* database service type specific arguments. Structure is documented below.",
+	Attributes: map[string]schema.Attribute{
+		"fork_from_service": schema.StringAttribute{
+			MarkdownDescription: "❗ Service name",
```
Comment (jessicatoscani) [inline pkg/resources/database/resource_opensearch.go:60]:
```suggestion
			MarkdownDescription: "Service name",
```

File: pkg/resources/database/resource_opensearch.go [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+			MarkdownDescription: "Maximum number of indexes to keep (Minimum value is `0`)",
+			Optional:            true,
+		},
+		"settings": schema.StringAttribute{
+			MarkdownDescription: "OpenSearch-specific settings, in json. e.g.`jsonencode({thread_pool_search_size: 64})`. Use `exo x get-dbaas-settings-opensearch` to get a list of available settings.",
+			Optional:            true,
+			Computed:            true,
+		},
+		"recovery_backup_name": schema.StringAttribute{
+			MarkdownDescription: "❗ Name of a backup to recover from",
```
Comment (jessicatoscani) [inline pkg/resources/database/resource_opensearch.go:89]:
```suggestion
			MarkdownDescription: "Name of a backup to recover from",
```

File: pkg/resources/database/resource_opensearch.go (hunk 1 above)
Comment (kobajagi) [reply pkg/resources/database/resource_opensearch.go:60]: That is deliberate. Documentation generator adds the symbol for `ForceReplace` in SDK, but for framework and PlanModifier (replacement for `ForceReplace`) it fails to detect it. I added the symbol to keep our documentation consistent.

File: pkg/resources/database/resource.go [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+			},
+			"nodes": schema.Int64Attribute{
+				MarkdownDescription: "The number of nodes of the database service.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+			},
+			"plan": schema.StringAttribute{
+				MarkdownDescription: "The plan of the database service (use the [Exoscale CLI](https://github.com/exoscale/cli/) - `exo dbaas type show <TYPE>` - for reference).",
```
Comment (jessicatoscani) [inline pkg/resources/database/resource.go]:
```suggestion
				MarkdownDescription: "The plan of the database service (use the [Exoscale CLI](https://github.com/exoscale/cli/) - `exo dbaas type show <TYPE> --plans` - for reference).",
```

File: pkg/resources/database/resource_opensearch.go (hunk 1 above)
Comment (jessicatoscani) [reply pkg/resources/database/resource_opensearch.go:60]: Ah true, I though it was `mandatory` :)

**Review by jessicatoscani — APPROVED** [review]

**Review by PhilippeChepy — APPROVED** [review]

File: pkg/resources/database/utils.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	ServicesList = []string{
```
Comment (sauterp) [inline pkg/resources/database/utils.go]: This seems like something that doesn't belong into a file called utils.go

**Review by sauterp — APPROVED** [review]

File: pkg/resources/database/utils.go (hunk 4 above)
Comment (kobajagi) [reply pkg/resources/database/utils.go]: Moved to `common.go`.

**Pull request merged**